File: Java/src/Spiral Matrix II.java
import java.util.Arrays;

public class Solution
{
  public int[][] generateMatrix(int n)
  {
    int[][] ret = new int[n][n];
    if (n == 0)
      return ret;
    if (n == 1)
    {
      ret[0][0] = 1;
      return ret;
    }

    for (int[] row : ret)
      Arrays.fill(row, 0);

    int up = 0;
    int down = n - 1;
    int left = 0;
    int right = n - 1;

    int data = 1;

    while (up <= down && left <= right)
    {
      for (int i = left; i <= right; i++)
      {
        ret[up][i] = data;
        data++;
      }
      up++;

      for (int i = up; i <= down; i++)
      {
        ret[i][right] = data;
        data++;
      }
      right--;

      for (int i = right; i >= left; i--)
      {
        ret[down][i] = data;
        data++;
      }
      down--;

      for (int i = down; i >= up; i--)
      {
        ret[i][left] = data;
        data++;
      }
      left++;

    }
    return ret;
  }
}

File: Python/Spiral Matrix II.py

class Solution:
    # @return a list of lists of integer
    def generateMatrix(self, n):
        if n == 0 :
            return []
        if n == 1:
            return [[1]]
        
        ret = [[0 for col in range(n)] for row in range(n)]
        
        up = 0
        down = n-1
        left = 0
        right = n-1

        i = 0
        j = 0
        data = 1
        while up <= down and left <= right:
            for i in range( left, right + 1):
                ret[up][i] = data
                data += 1
            up += 1
            
            for i in range(up, down + 1):
                ret[i][right] = data
                data += 1
            right -= 1
            
            for i in range(right, left-1, -1):
                ret[down][i] = data
                data += 1
            down -= 1
            
            for i in range(down , up - 1, -1):
                ret[i][left] = data
                data += 1
            left += 1
        return ret